File: src/main/java/Edge.java
public class Edge {
    String prefix;
    boolean isWord = true;
    MashupNode node;

    public Edge(String prefix, MashupNode node) {
        this.prefix = prefix;
        this.node = node;
    }

    @Override
    public String toString() {
        return "{" + "'" + prefix + "'" +
                ", " + Boolean.valueOf(isWord).toString().charAt(0) +
                (node.isEmpty() ? "" : ", " + node) +
                '}';
    }
}

File: src/main/java/Edge.kt
import java.lang.Boolean

internal class Edge(var prefix: String, var node: MashupNode) {
    @JvmField
    var isWord = true
    override fun toString(): String {
        return "{" + "'" + prefix + "'" +
                ", " + Boolean.valueOf(isWord).toString()[0] +
                (if (node.isEmpty) "" else ", $node") +
                '}'
    }
}